portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/PortalSiteHelperImpl.java

```diff
@@ -330,20 +344,77 @@ public String getGatewaySiteId()
 	}
 
 	/**
+	 * {@inheritDoc}
+	 */
+	public String getSiteOrSectionTitle( Site site )
+	{
+		return getSiteOrSectionTitle( site, getProviderIDsForSite( site ) );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getSiteOrSectionTitle( Site site, List<String> providerIDs )
+	{
+		String retVal = cms.determineSiteOrSectionTitle( site, providerIDs );
```

Comment:
Is it possible to not have CMS at all? It's part of edu-services by default. The way I understood it, was that there could be more than one impl, but by default there will be the CourseManagementServiceHibernateImpl (which has the actual implementation), CourseMangaemetnServiceSampleChainImpl, and CourseManagementServiceFederatedImpl.

The FederetedImpl loops through all known impls, and calls the method. In SampleChainImpl (which always seems to be the first impl in the list), the method throws an UnsupportedOperationException, which is caught inside the loop in FederatedImpl, then carries on to call the method on the next impl.

If all impls throw the exception, it falls back to returning the site title from the Site object passed into the method.

And due to the fact that the method is defined in the interface CourseManagementService, all classes implementing it would have to provide some sort of implementation of the method, even if it just throws an Unsupported exception.

So, I'm not sure how this could ever throw an NPE. Unless I'm mistaken and there is a way that you could deploy an instance without edu-services?